python/lsst/pipe/tasks/diff_matched_tract_catalog.py

```diff
@@ -685,13 +685,24 @@ def run(
         column_dist, column_dist_err = 'match_distance', 'match_distanceErr'
         dist = np.full(n_target, np.nan)
 
-        dist[matched_row] = np.hypot(
-            target.coord1[matched_row] - ref.coord1[matched_ref],
-            target.coord2[matched_row] - ref.coord2[matched_ref],
-        )
+        target_match_c1, target_match_c2 = (coord[matched_row] for coord in (target.coord1, target.coord2))
+        target_ref_c1, target_ref_c2 = (coord[matched_ref] for coord in (ref.coord1, ref.coord2))
+
         dist_err = np.full(n_target, np.nan)
-        dist_err[matched_row] = np.hypot(cat_target.iloc[matched_row][coord1_target_err].values,
-                                         cat_target.iloc[matched_row][coord2_target_err].values)
+        dist[matched_row] = sphdist(
+            target_match_c1, target_match_c2, target_ref_c1, target_ref_c2
+        ) if config.coord_format.coords_spherical else np.hypot(
+            target_match_c1 - target_ref_c1, target_match_c2 - target_ref_c2,
+        )
+        # Should probably explicitly add cosine terms if ref has errors too
+        dist_err[matched_row] = sphdist(
+            target_match_c1, target_match_c2,
+            target_match_c1 + cat_target.iloc[matched_row][coord1_target_err].values,
+            target_match_c2 + cat_target.iloc[matched_row][coord2_target_err].values,
+        ) if config.coord_format.coords_spherical else np.hypot(
```

Comment:
It is, but you raise a good point that I have to update the reference data because the spherical difference are different from the old hypot.

Just to confirm, only the distances change, i.e.:
`{k: v for k, v in dict(result.diff_matched.iloc[0] - self.diff_matched).items() if 'dist' not in k and (v != 0)}` prints nothing.

Reply:
I will add a second part to the test to set that config value to the opposite of the default and run again. Having said that, we don't generally test every single config permutation in tasks; it's only viable here since it's a boolean.